### PR TITLE
Remove check comparing shift/axis and input dimensions in np.roll

### PR DESCRIPTION
**Pull request by danielsuo**
Previously, `jax.numpy` checked that the number of shifts and/or axes was fewer than the number of dimensions of input. However, `numpy` doesn't make this check.

As a simple example, the following call will trigger a `ValueError`:
```python
import jax.numpy as np
np.roll([2, 3, 4, 5], (1, 0, 1))
```

However, `numpy` will treat a `shift` of `(1, 0, 1)` as though it were `np.sum(1, 0, 1)`.

We make the following changes:
1. Remove the dimensional check
2. Add two tests:
    a. Check when shift has high dimensionality
    b. Check that shift "overflow" works correctly (not really related to the issue at hand, but in the name of coverage)

### CODE REVIEW

**Comment by hawkinsp** [comment]
Thanks for the PR! Seems reasonable and the extra tests seem like good evidence this is the right thing to do.
